grpcurl.go

```diff
@@ -161,6 +163,36 @@ func MetadataFromHeaders(headers []string) metadata.MD {
 	return md
 }
 
+var envVarRegex = regexp.MustCompile(`\${\w+}`)
+
+// ExpandHeaders expands environmental variables contained in the header string.
+// If no corresponding environmental variable is found an error is thrown.
+// TODO: Add escaping for `${`
+func ExpandHeaders(headers []string) ([]string, error) {
+	expandedHeaders := make([]string, len(headers))
+	for idx, header := range headers {
+		if header != "" {
+			results := envVarRegex.FindAllString(header, -1)
+			if results != nil {
+				expandedHeader := header
+				for _, result := range results {
+					envVarName := result[2 : len(result)-1]
+					envVarValue := os.Getenv(envVarName)
+					replacementValue := envVarValue
+					if len(envVarValue) == 0 {
+						return nil, fmt.Errorf("environmental variable '%s' is not set", envVarName)
```

Comment:
It would be nice to reference the actual bad input. Maybe instead this:

```go
return nil, fmt.Errorf("header %q refers to missing environment variable %q", header, envVarName)
```